server/helpers/validation.ts

```diff
@@ -0,0 +1,70 @@
+import { plainToClass } from 'class-transformer';
+import { validate, ValidationSchema } from 'class-validator';
+import { Request, Response, NextFunction } from 'express';
+import { Post } from '../entities/post';
+
+// As the validation will be put to use on all the classes the structure of this file will need some refactoring.
+
+// conditionalValidation = isOptional -> https://github.com/typestack/class-validator/issues/147
+// The validation still doesn't check what the user inputs very well.
+export const postPUTSchema: ValidationSchema = {
+    name: "postPUTSchema",
+    properties: {
+        category: [{
+            type: 'conditionalValidation', constraints: [(object: any, value: any) => {
+                return object['message'] !== null && object['message'] !== undefined;
+            }]
+        },  {
+            type: "minLength",
+            constraints: [1]
+        }, {
+            type: "maxLength",
+            constraints: [50]
+        }, {
+            type: "isString"
+        }],
+        message: [{
+            type: 'conditionalValidation', constraints: [(object: any, value: any) => {
+                return object['message'] !== null && object['message'] !== undefined;
+            }]
+        }, {
+            type: "minLength",
+            constraints: [1]
+        }, {
+            type: "maxLength",
+            constraints: [1000]
+        }, {
+            type: "isString"
+        }]
+    }
+}
+
+export const validatePost = (req: Request, res: Response, next: NextFunction) => {
+    // Remove any extra fields and turn the plain object in to instance of a Post
+    req.body = plainToClass(Post, req.body, { excludeExtraneousValues: true })
```

Comment:
I would only modify `req.body` when this middleware was successful in validation

Reply:
Makes sense and adds readablity!